### PR TITLE
Set better default restart value

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/project-asgard/asgard/wiki/developing)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

Set a better default value of iterations before restart in gmres.

## What type(s) of changes does this code introduce?
_Put an `x` in the boxes that apply._

- [ ] Bugfix
- [x] New feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

### Does this introduce a breaking change?

- [ ] Yes
- [x] No

## What systems has this change been tested on?

Ubuntu 20.04

## Checklist

_Put an x in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- [x] this PR is up to date with current the current state of 'develop'
- [x] code added or changed in the PR has been clang-formatted
- [ ] this PR adds tests to cover any new code, or to catch a bug that is being fixed
- [ ] documentation has been added (if appropriate)

### CODE REVIEW

File: src/solver.cpp [hunk 1]
```diff
@@ -39,6 +41,18 @@ P simple_gmres(PDE<P> const &pde, elements::table const &elem_table,
   return simple_gmres(euler_operator, x, b, M, restart, max_iter, tolerance);
 }
 
+template<typename P>
+static int default_restart_value(int num_cols)
```
Comment (mkstoyanov) [inline src/solver.cpp]: A better name is `default_gmres_inner` and please put a doxygen clause on top of the method.

As a side note, I like the use of `clamp`

File: src/distribution.hpp [hunk 2]
```diff
@@ -266,7 +266,7 @@ double get_MB(int64_t const num_elems)
 {
   expect(num_elems > 0);
   double const bytes = num_elems * sizeof(P);
-  double const MB    = bytes * 1e-6;
+  double const MB    = bytes / 1048576;
```
Comment (mkstoyanov) [inline src/distribution.hpp:269]: Could this have been the source of the memory bug we saw on Tuesday?

File: src/solver.cpp (hunk 1 above)
Comment (mkstoyanov) [reply src/solver.cpp]: Maybe `default_gmres_restarts`

File: src/distribution.hpp (hunk 2 above)
Comment (quantumsteve) [reply src/distribution.hpp:269]: No, results should be very close. Just didn't like that it returned `mebibytes` instead of `megabytes`.

Comment (mkstoyanov) [reply src/distribution.hpp:269]: Actually I misread the result as `1e+6` as opposed to `-6`.

**Review by mkstoyanov — APPROVED** [review]

**Comment by mkstoyanov** [comment]
Looks great, merge if ready.

**Pull request merged**